code/_expts/slotnick_etal_2001/de_StimCreate.m

```diff
@@ -16,513 +16,432 @@
 %
 %  test.*     : same as train object, but
 
-if (~exist('opt','var')),      opt      = {};     end;
-if (~iscell(opt)),             opt      = {opt};  end;
-
-if (~strcmp(stimSet, 'blob-dot') && ~strcmp(stimSet, 'paired-squares') && ~strcmp(stimSet, 'plus-minus')) %default set of images
-    error('Invalid stimulus type. Choose one from: {''blob-dot'', ''paired-squares'', ''plus-minus''}');
-end
-
-%figure('Position', [0, 0, 1024, 800]); % is this line still needed?
-
-switch(stimSet)
-    case 'blob-dot',
-        % first, set some variables
-        
-        % Create the input images.
-        train.nInput = [68 50];
-        train.X = zeros(prod(train.nInput), 0);
-        train.XLAB = cell(20, 1); % blob-dot has 20 stimuli: 10 far/off, 5 near/off, 5 near/on
-        train.TLAB = cell(20, 1);
-        counter = 0;
-        heights = [];
-        
-        % first, 10 "far" images
-        for ii=1:5
-            distances = [10, 12];
-            for ij=1:length(distances)
-                img = blob_stimuli(distances(ij), 3, ii);
-                train.X(:, end+1) = reshape(img, prod(train.nInput), 1);
-                counter = counter + 1;
-                train.XLAB{counter} = sprintf('%dpx from %d%c', distances(ij), (ii-1)* 72, char(176));
-                heights = [heights, distances(ij)];
+    if (~exist('opt','var')),      opt      = {};     end;
+    if (~iscell(opt)),             opt      = {opt};  end;
+
+    switch(stimSet)
+        case 'blob-dot'
+            train = MakeBlobDot(taskType);
+
+        case 'paired-squares'
+            train = MakePairedSquares(taskType);
+
+        case 'plus-minus'
+            train = MakePlusMinus(taskType);
+
+        otherwise
+            error('Invalid stimulus type. Choose one from: {''blob-dot'', ''paired-squares'', ''plus-minus''}');
+    end;
+
+
+    % Now say that test data is the same as training data.
+    test = train;
+
+
+function train = MakeBlobDot(taskType)
+    if ~any(ismember(taskType, {'categorical', 'coordinate'}))
+        error('Unknown taskType: %s', taskType);
+    end;
+
+    % Create the input images.
+    train.nInput = [68 50];
+    train.X = zeros(prod(train.nInput), 0);
+    train.XLAB = {}; % blob-dot has 20 stimuli: 10 far/off, 5 near/off, 5 near/on
+    train.T = [];
+    train.TLAB = {};
+    heights = [];
+
+    distances = [0, 4, 8, 12];
+    ref_distance = 6;
+    for bi=1:5  % 5 blob images have been coded.
+        for d=distances
```

Comment:
@vishaalprasad Note that to iterate over each element of a vector, you can simply do: `       for d=distances`. This made the code much more succinct.